examples/providers/examples/builder.rs

```diff
@@ -0,0 +1,48 @@
+//! Example of using the `ProviderBuilder` to create a provider with a signer and network.
+
+use alloy_network::{Ethereum, EthereumSigner};
+use alloy_node_bindings::Anvil;
+use alloy_primitives::{U256, U64};
+use alloy_provider::{Provider, ProviderBuilder, RootProvider};
+use alloy_rpc_client::RpcClient;
+use alloy_rpc_types::TransactionRequest;
+use alloy_signer_wallet::Wallet;
+use alloy_transport_http::Http;
+use eyre::Result;
+use reqwest::Client;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // Setup the HTTP transport which is consumed by the RPC client
+    let anvil = Anvil::new().spawn();
+
+    let pk = &anvil.keys()[0];
+    let from = anvil.addresses()[0];
+    let signer = Wallet::from(pk.to_owned());
+
+    let rpc_client = RpcClient::new(Http::<Client>::new(anvil.endpoint().parse().unwrap()), false);
+    let provider_with_signer = ProviderBuilder::<_, Ethereum>::new()
+        .signer(EthereumSigner::from(signer))
+        .network::<Ethereum>()
+        .provider(RootProvider::new(rpc_client));
```

Comment:
Ideal UX here is `alloy_network::Ethereum::builder().signer(signer).connect(anvil.endpoint())`. We should strive to get to that, incl the gas / nonce filling layer being on by default with the option to turn it off with `.manual()` or something. cc @onbjerg @DaniPopes

Reply:
For now `::<_, Ethereum>` and `.network` are not necessary anymore: https://github.com/alloy-rs/alloy/pull/304